test/e2e/unmanaged_dns_test.go

```diff
@@ -0,0 +1,380 @@
+	t.Parallel()
+
+	name := types.NamespacedName{Namespace: operatorNamespace, Name: "managed-migrated"}
+	ic := newLoadBalancerController(name, name.Name+"."+dnsConfig.Spec.BaseDomain)
+	ic.Spec.EndpointPublishingStrategy.LoadBalancer = &operatorv1.LoadBalancerStrategy{
+		Scope:               operatorv1.ExternalLoadBalancer,
+		DNSManagementPolicy: operatorv1.ManagedLoadBalancerDNS,
+	}
+	if err := kclient.Create(context.TODO(), ic); err != nil {
+		t.Fatalf("failed to create ingresscontroller: %v", err)
+	}
+	defer assertIngressControllerDeleted(t, kclient, ic)
+
+	// Wait for the load balancer and DNS to reach stable conditions.
+	if err := waitForIngressControllerCondition(t, kclient, 5*time.Minute, name, availableConditionsForIngressControllerWithLoadBalancer...); err != nil {
+		t.Fatalf("failed to observe expected conditions: %v", err)
+	}
+
+	lbService := &corev1.Service{}
+	if err := kclient.Get(context.TODO(), controller.LoadBalancerServiceName(ic), lbService); err != nil {
+		t.Fatalf("failed to get LoadBalancer service: %v", err)
+	}
+
+	if ingresscontroller.IsServiceInternal(lbService) {
+		t.Fatalf("load balancer %s is internal but should be external", lbService.Name)
+	}
+
+	wildcardRecordName := controller.WildcardDNSRecordName(ic)
+	wildcardRecord := &iov1.DNSRecord{}
+	if err := kclient.Get(context.TODO(), wildcardRecordName, wildcardRecord); err != nil {
+		t.Fatalf("failed to get wildcard dnsrecord %s: %v", wildcardRecordName, err)
+	}
+
+	if wildcardRecord.Spec.DNSManagementPolicy != iov1.ManagedDNS {
+		t.Fatalf("DNSRecord %s expected in ManagedDNS state but got %s", wildcardRecordName.Name, wildcardRecord.Spec.DNSManagementPolicy)
+	}
+
+	verifyUnmanagedIngressController(t, name, "apps."+ic.Spec.Domain, wildcardRecord.Spec.Targets[0])
+
+	t.Logf("Updating ingresscontroller %s to Unmanaged", ic.Name)
+
+	if err := updateIngressControllerSpecWithRetryOnConflict(t, name, 5*time.Minute, func(ics *operatorv1.IngressControllerSpec) {
+		ics.EndpointPublishingStrategy.LoadBalancer.DNSManagementPolicy = operatorv1.UnmanagedLoadBalancerDNS
+	}); err != nil {
+		t.Fatalf("failed to update ingresscontroller %s: %v", name, err)
+	}
+
+	t.Log("Waiting for stable conditions on ingresscontroller after Unmanaged")
+
+	// Wait for the load balancer and DNS to reach stable conditions.
+	if err := waitForIngressControllerCondition(t, kclient, 10*time.Minute, name, availableConditionsForIngressControllerWithLoadBalancerUnmanagedDNS...); err != nil {
+		t.Fatalf("failed to observe expected conditions: %v", err)
+	}
+
+	if err := kclient.Get(context.TODO(), wildcardRecordName, wildcardRecord); err != nil {
+		t.Fatalf("failed to get wildcard dnsrecord %s: %v", wildcardRecordName, err)
+	}
+
+	if wildcardRecord.Spec.DNSManagementPolicy != iov1.UnmanagedDNS {
+		t.Fatalf("DNSRecord %s expected in UnmanagedDNS state but got %s", wildcardRecordName.Name, wildcardRecord.Spec.DNSManagementPolicy)
+	}
+
```

Comment:
So the enhancement specificially stated:

> - Test the following update paths on a custom ingresscontroller
>   - Updating `.loadbalancer.dnsManagementPolicy` from `Managed` -> `Unmanaged` -> `Managed` and
>     to ensure no conflicts during creation/recreation of DNS record on the cloud provider.

I don't think that whole test (Managed -> Unmanaged -> Managed) has exactly been accomplished yet. I'm hesitant to hold you to the exact words, but is it worth going back to Managed one last time here?

Reply:
So my initial thought was to add this transition and along with it manually delete the `DNS` records that are created on the cloud provider. But this would require a lot of additional test code to delete the records across cloud providers with very little returns (thoughts?). So without this logic, the last transition from `Unmanaged` -> `Transition` doesn't really make sense. 

So I'm thinking of just updating the EP w.r.t testing strategy along with the follow up change that is discussed [here](https://github.com/openshift/cluster-ingress-operator/pull/781#discussion_r929852173).

Reply:
That's a fair point. I think it would be good to keep the EP aligned with we are actually doing.